src/components/PostDetails.tsx

```diff
@@ -1,99 +1,53 @@
 import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  fetchComments,
+  deleteComment,
+  addComment,
+} from '../features/comments/commentsSlice';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
-
-import * as commentsApi from '../api/comments';
-
 import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
+import { CommentData } from '../types/Comment';
+import { RootState, AppDispatch } from '../app/store';
 
 type Props = {
   post: Post;
 };
 
 export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const dispatch = useDispatch<AppDispatch>();
+  const {
+    items: comments,
+    loaded,
+    hasError,
+  } = useSelector((state: RootState) => state.comments);
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi
-      .getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
+  useEffect(() => {
+    if (post) {
+      dispatch(fetchComments(post.id));
+    }
 
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
     setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
+  }, [dispatch, post]);
 
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
-    }
+  const handleAddComment = async (data: CommentData): Promise<void> => {
+    await dispatch(addComment({ postId: post.id, data }));
   };
 
-  useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
-
-  const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
-    }
+  const handleDeleteComment = async (commentId: number): Promise<void> => {
+    dispatch(deleteComment(commentId));
```

Comment:
Consider adding error handling logic inside the `handleDeleteComment` function to provide feedback to the user in case of failure.